collector/cfappscollector.go

```diff
@@ -37,6 +95,154 @@ func (c *CfAppsCollector) GetAnnotations() []string {
+		query.Set(inlineDepthRelations, queryDepth)
+		logrus.Debugf("CF collect query depth = %s", queryDepth)
+	} else {
+		query.Set(inlineDepthRelations, depth)
+		logrus.Debugf("CF collect query depth = %s", depth)
+	}
+	return query
+}
+
+//listAppsBySpaceGuid collects all CF apps data for given spaceGuid, if no appName/appGuid is present. If appName/appGuid is present, then it updates the path to be used in the listAppsByNameOrGuid func
+func listAppsBySpaceGuid(client *cfclient.Client, spaceGuid string, queryDepth string, collectApps []App, numCfCollectApps int) (string, []App) {
+	query := setQueryDepth(queryDepth)
+	path := ""
+	if spaceGuid != "" {
+		logrus.Debugf("Detected CF Space guid: %s", spaceGuid)
+		path = fmt.Sprintf("/v2/spaces/%s/apps", spaceGuid)
+		if numCfCollectApps == 0 {
+			apps, err := listApps(client, path, query, -1) // If no CF app is specified in yaml, collect all apps in the provided spaceguid
+			if err != nil {
+				logrus.Errorf("Unable to collect the cf apps from the Space guid %s : %s", spaceGuid, err)
+			} else {
+				collectApps = append(collectApps, apps...)
+			}
+		}
+	}
+	return path, collectApps
+}
+
+func getAppByGuid(client *cfclient.Client, guid string, queryDepth string) (App, error) {
+	var appResource AppResource
+	query := setQueryDepth(queryDepth)
+	requestUrl := getRequestUrl("/v2/apps/"+guid, query) // /v2/apps/:appGuid fetches the app with given guid
+	logrus.Debugf("CF collect queryDepth val = %v", queryDepth)
+	logrus.Debugf("CF collect request URL = %v", requestUrl)
+	r := client.NewRequest("GET", requestUrl)
+	resp, err := client.DoRequest(r)
+	if err != nil {
+		return App{}, errors.Wrap(err, "Error requesting apps")
+	}
+	defer resp.Body.Close()
+	resBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return App{}, errors.Wrap(err, "Error reading app response body")
+	}
+	err = json.Unmarshal(resBody, &appResource)
+	if err != nil {
+		return App{}, errors.Wrap(err, "Error unmarshalling app")
+	}
+	return mergeAppResource(client, appResource), nil
+}
+
+func getAppsByNameOrGuid(client *cfclient.Client, path string, query url.Values, collectApps []App, appName string, appGuid string) []App {
+	apps, err := listApps(client, path, query, -1)
+	if err != nil {
+		logrus.Errorf("Unable to collect the selected cf app %s %s : %s", appName, appGuid, err)
+	} else {
+		if len(apps) == 0 {
+			cfErr := cfclient.NewAppNotFoundError()
+			if appGuid != "" {
+				logrus.Errorf(fmt.Sprintf(cfErr.Description, appGuid))
+			} else {
+				logrus.Errorf(fmt.Sprintf(cfErr.Description, appName))
+			}
+		} else {
+			collectApps = append(collectApps, apps...)
+		}
+	}
+	return collectApps
+}
+
+func listAppsByNameOrGuid(client *cfclient.Client, path string, cfCollectApps []CfCollectApp, queryDepth string) []App {
+	var collectApps []App
+	for _, cfCollectApp := range cfCollectApps {
+		if path == "" {
+			path = listAppsPath
+		}
+		appSpec := cfCollectApp.AppSpec
+		if appSpec.Guid != "" {
```

Comment:
I have update the code. First we check if app Guid is given, if the app data is collected successfully for the app Guid, then we `continue` the for loop for the next app. Otherwise, we check if app Name is given and try to collect the data of the app. 

Also, I have made changes in the code to use shorter return in some other places too.